### PR TITLE
Add input validation for save_checkpoint and warmup LR schedulers

### LINKED ISSUE #8097: Missing input validation could cause unexpected behavior with edge case inputs
### Description
Some functions do not validate their inputs before processing, which can lead to confusing error messages or silent failures when unexpected values are passed.

### Steps to Reproduce
1. Call the affected function with `None`, an empty string, or a negative number
2. Observe that the error message is not helpful or the function silently returns incorrect results

### Expected Behavior
Functions should validate inputs early and raise descriptive errors:

```python
def process(value):
    if value is None:
        raise ValueError("value must not be None")
    if not isinstance(value, int):
        raise TypeError(f"expected int, got {type(value).__name__}")
    # ... rest of function
```

### Impact
This makes debugging harder for users of the library and can hide real bugs upstream.

### Suggested Fix
Add input validation at the start of public-facing functions with clear error messages.

### PR DESCRIPTION
**Pull request by jahnavi-yelamanchi**
## Summary
Fixes #8097. Some public-facing functions silently accept invalid input (None, empty string, negative numbers) leading to confusing downstream errors or hangs. This PR adds explicit validation in two places:

- `DeepSpeedEngine.save_checkpoint`: raises `ValueError` if `save_dir` is None/empty, before any rank-synchronization logic runs (prevents multi-GPU hang from partial-rank failure).
- `WarmupLR.__init__` / `WarmupCosineLR.__init__`: raises `ValueError` if `warmup_num_steps` isn't a positive int. `WarmupDecayLR` inherits the check via `WarmupLR`.

## Test plan
- [x] `pytest tests/unit/runtime/test_lr_schedulers.py -k warmup` — new parametrized test passes for all 3 scheduler classes
- [x] `pytest --forked tests/unit/checkpoint/test_tag_validation.py` — new test added; skips locally (needs 2 GPUs, unavailable on this machine), expected to run in CI
- [x] `pre-commit run` on all changed files — clean

### CODE REVIEW

**Review by tohtana — APPROVED** [review]
Hi @jahnavi-yelamanchi,
Looks good to me, thank you for your contribution!
